core/unix/os.c

```diff
@@ -7058,6 +7062,10 @@ pre_system_call(dcontext_t *dcontext)
             execute_syscall = false;
             break;
         }
+        if (data.sigmask == NULL) {
+            dcontext->sys_param3 = (reg_t)NULL;
```

Comment:
nit: This seems confusing to me to have the params mean different things: seems better to consistently use sys_param4 which is set just one line below and could be moved above here

Reply:
Done.